props/saber_fett263_buttons.h

```diff
@@ -845,35 +5080,86 @@ SaberFett263Buttons() : PropBase() {}
           beeper.Beep(0.5, 3000);
         }
         return;
+#ifndef FETT263_SAY_BATTERY
       case EFFECT_BATTERY_LEVEL:
         if (SFX_battery) {
           hybrid_font.PlayCommon(&SFX_battery);
         } else {
           beeper.Beep(0.5, 3000);
         }
         return;
+#endif
       case EFFECT_FAST_ON:
         if (SFX_faston) {
           hybrid_font.PlayCommon(&SFX_faston);
         }
         return;
-      default: break; // avoid compiler warnings
+      default:
+        break; // avoid compiler warnings
     }
   }
 
 private:
-  bool pointing_down_ = false;
-  bool swing_blast_ = false;
-  bool mode_volume_ = false;
-  bool auto_lockup_on_ = false;
-  bool auto_melt_on_ = false;
-  bool battle_mode_ = false;
-  uint32_t thrust_begin_millis_ = millis();
-  uint32_t push_begin_millis_ = millis();
-  uint32_t clash_impact_millis_ = millis();
-  uint32_t last_twist_ = millis();
-  uint32_t last_push_ = millis();
-  uint32_t saber_off_time_ = millis();
+  bool swing_blast_ = false; // Multi-Blast Mode (Swing = Blast Deflect)
+  bool check_blast_ = false; // Check if Multi-Blast mode should be enabled (Battle Mode)
+  bool auto_lockup_on_ = false; // Battle Mode Lockup active
+  bool auto_melt_on_ = false; // Battle Mode Melt/Drag active
+  bool battle_mode_ = false; // Battle Mode active
+  bool menu_ = false; // Edit Mode / Menu System active
+  bool force_quote_ = false; // Quote Player active (in place of force effect)
+  bool rehearse_ = false; // Rehearsal Mode active
+  bool choreo_ = false; // Choreography Mode active
+  uint32_t thrust_begin_millis_;
+  uint32_t push_begin_millis_;
+  uint32_t clash_impact_millis_;
+  uint32_t last_twist_;
+  uint32_t last_push_;
+  uint32_t last_blast_;
+  uint32_t saber_off_time_;
+  uint32_t last_rotate_;
+  ClashType clash_type_ = CLASH_NONE;
+  MenuType menu_type_ = MENU_TOP;
+  int menu_top_pos_ = 0;
+  int menu_sub_pos_ = 0;
+  TrackMode track_mode_ = PLAYBACK_OFF;
+  int track_num_;
+  int num_tracks_;
+  int ignite_time_;
+  int dial_ = -1;
+  int sub_dial_;
+  float change_ = 0.0;
+  float twist_menu_ = M_PI / 4;
+#ifdef FETT263_EDIT_MODE_MENU
+  bool choice_ = false; // Edit Mode selection confirmed
+  bool next_event_ = false; // Do next event in Edit Mode
+  bool off_event_ = false; // Do off event in Edit Mode
+  bool restart_ = false; // Ignite blade after off event in Edit Mode
+  bool edit_color_ = false; // Color Editing Mode active
+  uint32_t color_revert_;
```

Comment:
All the variables below needs comments too.
(and there are more above that needs comments.)

Reply:
added